src/genericfit.jl

```diff
@@ -30,5 +30,15 @@ fit_mle(dt::Type{D}, x::AbstractArray, w::AbstractArray) where {D<:UnivariateDis
 fit_mle(dt::Type{D}, x::AbstractMatrix) where {D<:MultivariateDistribution} = fit_mle(D, suffstats(D, x))
 fit_mle(dt::Type{D}, x::AbstractMatrix, w::AbstractArray) where {D<:MultivariateDistribution} = fit_mle(D, suffstats(D, x, w))
 
+"""
+    fit(D, x)
+
+Fit a distribution of type `D` to a dataset `x` using an arbitrary algorithm.
```

Comment:
I guess
```suggestion
Fit a distribution of type `D` to a dataset `x`.
```
would be sufficient?

Additionally, I think the term `dataset` is very unclear - sounds a bit like it should be a data frame or table even though usually just vectors of samples are suppported.